avr-hal-generic/src/i2c.rs

```diff
@@ -217,23 +228,19 @@ macro_rules! impl_twi_i2c {
             scl: $sclmod::$SCL<M>,
         }
 
-        impl<CLOCK, M> [<$I2c Master>]<CLOCK, M>
-        where
-            CLOCK: $crate::clock::Clock,
-        {
-            fn initialize_i2c(
```

Comment:
Added a static method for initializing TWI.

Reply:
This is a good idea!  Though maybe put it into the existing impl block further down instead of adding a second one up top?  You could also take the pattern from the `serial` module and make the initializer a method instead of a static function:

https://github.com/Rahix/avr-hal/blob/7d851d50b2485257a2cbede991e8de4c734053d8/avr-hal-generic/src/serial.rs#L139-L173

Reply:
Cool, will do.